### PR TITLE
DOC: GL01 numpydoc validation

### LINKED ISSUE #25324: Fix validation error type `GL01` and check in CI
#### Problem description
In order to have a continuous check by the CI on validation error `GL01` (Docstring text, i.e. summary, should start in the line immediately after the opening quotes, not in the same line, or leaving a blank line in between), fixing them in the code base enables the addition to the CI for automated testing in the future.

`GL01` has 189 occurrences (see [overview of the errors](https://gist.github.com/198cfa552c727d71bd7360fda66fdbb4)), including:

method | file | file_line | github_link
-- | -- | -- | -- 
pandas.Timestamp | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.max | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.min | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.ctime | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.date | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.dst | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.isocalendar | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.isoweekday | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.strftime | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.strptime | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.time | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.timestamp | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.timetuple | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.timetz | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.to_datetime64 | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.toordinal | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.tzname | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.utcoffset | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.utctimetuple | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.weekday | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timedelta.components | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timedelta.days | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timedelta.microseconds | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timedelta.seconds | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timedelta.view | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timedelta.to_timedelta64 | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Interval.left | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Interval.length | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Interval.right | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Categorical.codes | pandas/core/arrays/categorical.py | 676.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/arrays/categorical.py#L676
... | ... | ... | ...
pandas.plotting.lag_plot | pandas/plotting/_misc.py | 570.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/plotting/_misc.py#L570
pandas.plotting.parallel_coordinates | pandas/util/_decorators.py | 454.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/util/_decorators.py#L454
pandas.core.resample.Resampler.pipe | pandas/core/resample.py | 189.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/resample.py#L189
pandas.core.resample.Resampler.first | pandas/core/resample.py | 857.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/resample.py#L857
pandas.core.resample.Resampler.last | pandas/core/resample.py | 857.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/resample.py#L857
pandas.core.resample.Resampler.max | pandas/core/resample.py | 857.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/resample.py#L857
pandas.core.resample.Resampler.min | pandas/core/resample.py | 857.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/resample.py#L857
pandas.core.resample.Resampler.prod | pandas/core/resample.py | 846.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/resample.py#L846
pandas.core.resample.Resampler.sum | pandas/core/resample.py | 846.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/resample.py#L846
pandas.Series.index | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Series.T | pandas/core/base.py | 672.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/base.py#L672
pandas.Series.dt.dayofweek | pandas/core/accessor.py | 80.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/accessor.py#L80
pandas.Series.dt.weekday | pandas/core/accessor.py | 80.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/accessor.py#L80
pandas.Series.dt.is_month_start | pandas/core/accessor.py | 80.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/accessor.py#L80
pandas.Series.dt.is_month_end | pandas/core/accessor.py | 80.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/accessor.py#L80
pandas.Series.dt.is_quarter_start | pandas/core/accessor.py | 80.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/accessor.py#L80
pandas.Series.dt.is_quarter_end | pandas/core/accessor.py | 80.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/accessor.py#L80
pandas.Series.dt.is_year_start | pandas/core/accessor.py | 80.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/accessor.py#L80
pandas.Series.dt.is_year_end | pandas/core/accessor.py | 80.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/accessor.py#L80
pandas.Series.dt.is_leap_year | pandas/core/accessor.py | 80.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/accessor.py#L80
pandas.io.formats.style.Styler.env | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.io.formats.style.Styler.template | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.io.formats.style.Styler.loader | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Series.ix | pandas/core/indexing.py | 1385.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/indexing.py#L1385
pandas.DataFrame.ix | pandas/core/indexing.py | 1385.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/indexing.py#L1385
pandas.Panel.items | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Panel.ix | pandas/core/indexing.py | 1385.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/indexing.py#L1385
pandas.Panel.keys | pandas/core/generic.py | 1824.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/core/generic.py#L1824
pandas.Panel.major_axis | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Panel.minor_axis

Todo:

* [ ]    get rid of the errors in the code base
* [ ]    update the `code_check.sh` script to take into account the `GL01` type of errors

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #25324
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

### CODE REVIEW

File: pandas/io/formats/style_render.py [hunk 1]
```diff
@@ -57,17 +57,23 @@ class CSSDict(TypedDict):
 Subset = Union[slice, Sequence, Index]
 
 
+def _gl01_adjust(obj: Any) -> Any:
```
Comment (jreback) [inline pandas/io/formats/style_render.py:60]: umm this seems kind of a hack, can we just adjust the templates?

Comment (mroeschke) [reply pandas/io/formats/style_render.py:60]: These are the docstrings of `jinja2.PackageLoader` and `jinja2.Environment` I believe, and their docstrings don't conform with `GL01` cc @attack68

Comment (attack68) [reply pandas/io/formats/style_render.py:60]: thats right these are inherited docstring objects.

is there a way we can prevent the doc string validation is such cases? might be better than overwriting it?

Comment (mroeschke) [reply pandas/io/formats/style_render.py:60]: Thanks. Looks like the validation script doesn't have any capabilities to ignore certain objects or methods; it looks to check everything in the public API.

Comment (jreback) [reply pandas/io/formats/style_render.py:60]: ok merging

but maybe would be good to enable some ignoring to prevent having to do this

**Review by jreback — APPROVED** [review]

**Pull request merged**